### ISSUE TITLE
I have few question.

### ISSUE THREAD

**Issue by jeremywrightkim** [lck1201/hello-world#2]
I'm interest in your removed `cascadehandpose` project.
I already have MSRA dataset. But I don't know input data information.
For example `Path_Images.txt` and `.pts `files ets...
Perhaps the image/data being loaded is different from the existing data.
Can you send me a dataset by email (jeremywrightkim@gmail.com)?
or dropbox, google drive etc...?
Please give detailed explanation. 
I really appreciate you. thank you

**Comment by lck1201** [comment]
check your email box

**Comment by jeremywrightkim** [comment]
Ok. 
I have not received a e-mail yet.
I will be waiting. thank you. 

jeremywrightkim@gmail.com

**Comment by jeremywrightkim** [comment]
`ConstructPointCloud()` function use `Mat img`.
I think it is a depth image.
But, MSRAdataset only have `*.bin files` which include depth value in ROI of hand.
MSRAdataset does not provide depth images.

So, I try to make a depth image using *.bin files.
But I don't know background value and image type you use.
Is image type `CV_16UC1` ? What is the background value?
What does threshold value 1000 mean? Is background value greater than 1000 ?
Please explain how to create a depth image that you use.

Thank you for your precious time.

```
for (int i = 0; i < depthmap.rows; i++){ // for each row
	for (int j = 0; j < depthmap.cols; j++){ // for each col
		if (depthmap.at<ushort>(i, j) < 1000){ // set a threshold
			double *data = PointCloud.ptr<double>(flag++);				
			cout << PointCloud.at<double>(i, j);
			data[0] =  j - cx;
			data[1] =  i - cy;
			data[2] =  depthmap.at<ushort>(i, j);	
			cout << PointCloud.at<double>(i, j);
		}
			
	}
}
```

**Comment by jeremywrightkim** [comment]
I'm sorry for many questions.
I made `.pts` and `.png` files as you told me. thank you.

I called functions in the order of `prepare ()` > `train ()` > `test ()`.
But, the visual results show many errors.

![18](https://cloud.githubusercontent.com/assets/26728351/24768267/f3a7e1ce-1b3c-11e7-8d9a-7b9524eb89b9.jpg)
![0](https://cloud.githubusercontent.com/assets/26728351/24768269/f3b4e694-1b3c-11e7-89c1-c1e399306584.jpg)

What's the problem?
Did I create the data incorrectly? Are there less learning sets?
I don't know the cause of this problem.

I used a subset of the whole data for training.
My subset is consist of 500 data in `cvpr15_MSRAHandGestureDB\P0\1`
It is in the link below.

https://www.dropbox.com/sh/afgvpeu484b4ejv/AAAmNVRl5YaaIbi_2zNBtq3ga?dl=0

**Comment by jeremywrightkim** [comment]
You can download full raw set in my dropbox link. 

https://www.dropbox.com/sh/5iukb2es5zjh6b3/AADy-cnrgJ22c7Lg05uPGCxNa?dl=0

**Comment by jeremywrightkim** [comment]
1) Can you tell me what kind of problem you have?
2) Can you provide me with learned 7 type of .model data?

Estimation results I've trained using dataset show similar approximations but are not accurate.

![21](https://cloud.githubusercontent.com/assets/26728351/25065253/65d0f5b8-2247-11e7-80dd-277f2dffc871.jpg)
![40](https://cloud.githubusercontent.com/assets/26728351/25065258/6f6e1682-2247-11e7-8800-a4ab5dbc5696.jpg)

**Comment by jeremywrightkim** [comment]
Is it difficult to give an answer? :(

**Comment by jeremywrightkim** [comment]
I spent a lot of time for training, but I am having memory related problems, and I do not know the cause. I am analyzing the your code, but no memory leaks have been found in your code. 
Currently, when training with less than 2000 training data for a single pose, the training process completes without memory problems, but the results are not accurate.
When training with more than 2000 training data, the application is interrupted.
The amount of 2000 data is insufficient to train multiple poses.

Probably because my computer has a small amount of memory. 
(My PC uses 8GB RAM. Additional ordering 8GBRAM being shipped)

So I was asking if you could provide the learned .model file about multiple poses.
(Palm.model, index_finger.model, mid_finger.model, ring_finger.model, little_finger.model, thumb_finger.model)

That's because I wanted to see the learned results of multiple poses.
I will solve the accuracy problems.  

I am sorry for many questions.
Thank you.

**Comment by jeremywrightkim** [comment]
I read the paper in detail as you said.
Theoretically, I understood the whole contents of the paper.
Then, I've looked at your code based on understanding of the paper, but I do not know the problem. :(
I think it's probably a `palm or finger canonical transformation matrix` related problem.
Or a `random forest` problem? or `hand scale of training data.`..?
I have not looked at random forest source code yet. It is complicated.
I'm sorry I could not find the problem. I will try to look into it in more detail.

**Comment by jeremywrightkim** [comment]
Hello. Long time no see.
I've looked at the full source code detail, but I do not know the problem.

I think that there is no problem in the random forest training part.

I thought it was a transformation problem between 3D feature extraction and 2D feature extraction.
However, after debugging, I could not find any problems.
But, I still think the problem is a feature related problem. It seems to be a difficult problem for me.

Palm joints are relatively well classified, but finger joints contain many errors. Especially the thumb..
Any hints or tips can you give me? I'm sorry to bother you.

![36](https://cloud.githubusercontent.com/assets/26728351/25838303/683506d0-34cd-11e7-9f5a-fe8400d22727.jpg)
![37](https://cloud.githubusercontent.com/assets/26728351/25838305/6bcc295e-34cd-11e7-98ec-970d190986b5.jpg)
![24](https://cloud.githubusercontent.com/assets/26728351/25838287/5817440c-34cd-11e7-89f3-fe7a561a1d07.jpg)
![31](https://cloud.githubusercontent.com/assets/26728351/25838293/5d2b4ede-34cd-11e7-87fe-a0f27bd820b6.jpg)

**Comment by jeremywrightkim** [comment]
Thank you for your advice. I understand your advice, check the issue, and try to fix it.
